### PR TITLE
Bump kube-rbac-proxy version for kserve-controller

### PR DESCRIPTION
**Pull request by ddelange**
**What this PR does / why we need it**:

Add support for running controller on arm64 xref https://github.com/kserve/kserve/pull/2550, https://github.com/kserve/kserve/pull/2627

```console
$ docker run --rm mplatform/mquery gcr.io/kubebuilder/kube-rbac-proxy:v0.13.1
Image: gcr.io/kubebuilder/kube-rbac-proxy:v0.13.1 (digest: sha256:d4883d7c622683b3319b5e6b3a7edfbf2594c18060131a8bf64504805f875522)
 * Manifest List: Yes (Image type: application/vnd.docker.distribution.manifest.list.v2+json)
 * Supported platforms:
   - linux/amd64
   - linux/arm64
   - linux/ppc64le
   - linux/s390x
```

**Which issue(s) this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close the issue(s) when PR gets merged)*:
Fixes #

**Type of changes**
Please delete options that are not relevant.

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update

**Feature/Issue validation/testing**:

Please describe the tests that you ran to verify your changes and relevent result summary. Provide instructions so it can be reproduced.
Please also list any relevant details for your test configuration.

- [ ] Test A
- [ ] Test B

- Logs

**Special notes for your reviewer**:

1. Please confirm that if this PR changes any image versions, then that's the sole change this PR makes.

**Checklist**:

- [ ] Have you added unit/e2e tests that prove your fix is effective or that this feature works?
- [ ] Has code been commented, particularly in hard-to-understand areas?
- [ ] Have you made corresponding changes to the documentation?

**Release note**:

```release-note

```

### CODE REVIEW

**Comment by ddelange** [comment]
running [`dockersize`](https://stackoverflow.com/a/73108928/5511061) on the [first party](https://github.com/brancz/kube-rbac-proxy/blob/v0.14.0/.github/workflows/build.yml#L101) image shows that this one additionally gets built for 32bit arm by the way (adding support for it in the linked PRs above, for [feature parity](https://github.com/kserve/kserve/pull/2550#discussion_r1055222609) with knative):
```console
$ dockersize quay.io/brancz/kube-rbac-proxy:v0.14.0
linux/amd64    23.51M
linux/arm      22.34M
linux/arm64    21.89M
linux/ppc64le  21.69M
linux/s390x    23.22M
```

any particular reason to use the `kubebuilder` image instead? it seems to be behind on latest release

File: charts/kserve-resources/templates/deployment.yaml [hunk 1]
```diff
@@ -25,7 +25,7 @@ spec:
       serviceAccountName: kserve-controller-manager
       containers:
       - name: kube-rbac-proxy
-        image: gcr.io/kubebuilder/kube-rbac-proxy:v0.4.0
```
Comment (yuzisun) [inline charts/kserve-resources/templates/deployment.yaml:28]: @ddelange Can you also help update the version in helm chart?
https://github.com/kserve/kserve/blob/master/charts/kserve-resources/templates/deployment.yaml#L28

Comment (ddelange) [reply charts/kserve-resources/templates/deployment.yaml:28]: the line you posted is the line which this PR updates. which version do you mean to update exactly? the `Chart.yaml` version bump is part of your release process I thought? so I didn't touch it. Would you like me to increase it to rc2?

Comment (yuzisun) [reply charts/kserve-resources/templates/deployment.yaml:28]: @ddelange I meant for making the same change for `kube-rbac-proxy` version in the helm chart not the chart version.

Comment (yuzisun) [reply charts/kserve-resources/templates/deployment.yaml:28]: We have two installation methods, the place you modified is for kustomize and we also support helm installation.

Comment (ddelange) [reply charts/kserve-resources/templates/deployment.yaml:28]: > the line you posted is the line which this PR updates

this PR updates the tag in the helm chart, can you link me to the line which additionally needs updating?

Comment (yuzisun) [reply charts/kserve-resources/templates/deployment.yaml:28]: @ddelange sorry my bad to give the wrong link, it should be this place 
https://github.com/kserve/kserve/blob/master/config/default/manager_auth_proxy_patch.yaml#L13

Comment (ddelange) [reply charts/kserve-resources/templates/deployment.yaml:28]: do you mean https://github.com/kserve/kserve/blob/19ee4cf428f852007b3ca979029de209fbf88ab9/config/default/manager_auth_proxy_patch.yaml#L13 ?

Comment (ddelange) [reply charts/kserve-resources/templates/deployment.yaml:28]: done!

Comment (ddelange) [reply charts/kserve-resources/templates/deployment.yaml:28]: ah damn DCO, github web ui didnt recognise it, sec

**Comment by yuzisun** [comment]
thanks @ddelange !

/lgtm
/approve

**Pull request merged**